docs/concepts.md

```diff
@@ -0,0 +1,136 @@
+
+[**Questions**](./question.md) are all independent, and a given question can be used on many different assessments across many different course instances. They are written in HTML (as a [Mustache template](https://mustache.github.io/mustache.5.html)) and use [elements](./elements.md) to accept student input. Python code can be used to generate random parameters and grade questions. Student code submissions can also be graded with an [external code autograder](./externalGrading.md).
+
+#### Example `question.html`
+
+```html title="question.html"
+<pl-question-panel>
+  If $x = {{params.x}}$ and $y$ is {{params.operation}} $x$, what is $y$?
+</pl-question-panel>
+
+<pl-number-input answers-name="y" label="$y =$"></pl-number-input>
+```
+
+#### Example `server.py`
+
+```python title="server.py"
+def generate(data):
+    data["params"]["x"] = random.randint(5, 10)
+    data["params"]["operation"] = random.choice(["double", "triple"])
+
+    if data["params"]["operation"] == "double":
+        data["correct_answers"]["y"] = 2 * data["params"]["x"]
+    else:
+        data["correct_answers"]["y"] = 3 * data["params"]["x"]
+```
+
+??? example "Question with custom partial credit & feedback"
+
+    === "Example `question.html`"
+
+        ```html title="question.html"
+        <pl-question-panel>
+            If $x = {{params.x}}$ and $y$ is {{params.operation}} $x$, what is $y$?
+        </pl-question-panel>
+
+        <pl-number-input answers-name="y" label="$y =$"></pl-number-input>
+        <pl-submission-panel> {{feedback.y}} </pl-submission-panel>
+        ```
+
+    === "Example `server.py`"
+
+        ```python title="server.py"
+        def generate(data):
+            data["params"]["x"] = random.randint(5, 10)
+            data["params"]["operation"] = random.choice(["double", "triple"])
+
+            if data["params"]["operation"] == "double":
+                data["correct_answers"]["y"] = 2 * data["params"]["x"]
+            else:
+                data["correct_answers"]["y"] = 3 * data["params"]["x"]
+
+        def parse(data):
+            if "y" not in data["format_errors"] and data["submitted_answers"]["y"] < 0:
+                data["format_errors"]["y"] = "Negative numbers are not allowed"
+
+        def grade(data):
+            if math.isclose(data["score"], 0.0) and data["submitted_answers"]["y"] > data["params"]["x"]:
+                data["partial_scores"]["y"]["score"] = 0.5
+                data["score"] = 0.5
+                data["feedback"]["y"] = "Your value for $y$ is larger than $x$, but incorrect."
+        ```
+
+### Elements
+
+Questions can use **elements** to accept student input, display diagrams, control how questions are displayed. Elements are pre-built interface elements for common questions, like numerical inputs, multiple choice, and more. If you require more flexibility -- **custom elements** are reusable components that can be tailored to your course and provide a way to create more complex questions.
```

Comment:
```suggestion
Questions can use **elements** to accept student input, display diagrams, control how questions are displayed. Elements are pre-built interface elements for common questions, like numerical inputs, multiple choice, and more. If you require more flexibility, you can build **custom elements** that can be tailored to your course and provide a way to create more complex questions.
```

Maybe link the phrase "custom elements" to the part of the docs that discusses them?